rdflib/term.py

```diff
@@ -1255,7 +1263,9 @@ def _quote_encode(self):
                 '\n', '\\n').replace(
                     '\\', '\\\\').replace(
                         '"', '\\"').replace(
-                            '\r', '\\r')
+                            '\r', '\\r').replace(
+                                "inf", "INF").replace(
+                                    "nan", "NaN")
```

Comment:
sorry, this escaped me on first review: this also replaces `"inf"` and `"nan"` in strings...
```python
In [4]: Literal("nananananananananananananananana BATMAAAN").n3()
Out[4]: u'"NaNaNaNaNaNaNaNaNaNaNaNaNaNaNaNa BATMAAAN"'
```